test/e2e/storage/testsuites/provisioning.go

```diff
@@ -261,16 +254,39 @@ func (t StorageClassTest) TestDynamicProvisioning() *v1.PersistentVolume {
 			framework.Failf("Error deleting claim %q. Error: %v", claim.Name, err)
 		}
 	}()
-	err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, claim.Namespace, claim.Name, framework.Poll, framework.ClaimProvisionTimeout)
```

Comment:
The original dynamic provisioning tests do not work for delayed binding because they're waiting for the PVC to be bound before creating pods.

To fix this, I've reversed the ordering. First, run PvCheck which creates the pods, and then afterwards verify the PVC and PV.